loaders/src/glTF/2.0/Extensions/KHR_mesh_instancing.ts

```diff
@@ -0,0 +1,111 @@
+                const attributeBufferPromises = new Array<Promise<any>>();
+
+                let attributes = extension.attributes;
+                let attributeKeys = Object.keys(attributes);
+                const attributeBuffers : { [name: string]: { buffer : Float32Array, accessor : IAccessor } } = {};
+                for (let i = 0; i < attributeKeys.length; ++i) {
+                    let attributeKey = attributeKeys[i];
+                    let accessor = ArrayItem.Get(`${extensionContext}/attributes/${attributeKey}`, this._loader.gltf.accessors, attributes[attributeKey]);
+                    attributeBufferPromises.push(this._loader._loadFloatAccessorAsync(`/accessors/${accessor.bufferView}`, accessor).then((data) => {
+                        attributeBuffers[attributeKey] = { buffer : data, accessor : accessor };
+                    }));
+                }
+
+                const mesh = ArrayItem.Get(`${context}/mesh`, this._loader.gltf.meshes, extension.mesh || nodeMesh);
+                this._loader._loadMeshAsync(`/meshes/${mesh.index}`, node, mesh, (babylonMeshInstanceNode) => {
+                    Promise.all(attributeBufferPromises).then(() => {
+                        attributeKeys = Object.keys(attributeBuffers);
+
+                        if (attributeKeys.length > 0) {
+                            let instanceCount = attributeBuffers[attributeKeys[0]].accessor.count;
+                            let digitLength = instanceCount.toString().length;
+                            for (let i = 0; i < instanceCount; ++i) {
+                                let instance;
+                                if (i == 0) {
+                                    instance = babylonMeshInstanceNode;
+                                    instance.setParent(babylonTransformNode);
+                                } else {
+                                    instance = babylonMeshInstanceNode.clone((babylonMeshInstanceNode.name || babylonMeshInstanceNode.id) + "_" + String(i).padStart(digitLength, '0'), babylonTransformNode, true);
+                                }
+
+                                if (instance) {
+                                    instance.position = attributeBuffers["TRANSLATION"] ? Vector3.FromArray(attributeBuffers["TRANSLATION"].buffer, i * 3)
+                                                                                        : Vector3.Zero();
+                                    instance.rotationQuaternion = attributeBuffers["ROTATION"] ? Quaternion.FromArray(attributeBuffers["ROTATION"].buffer, i * 4)
+                                                                                            : Quaternion.Identity();
+                                    instance.scaling = attributeBuffers["SCALE"] ? Vector3.FromArray(attributeBuffers["SCALE"].buffer, i * 3)
+                                                                                    : Vector3.One();
```

Comment:
Should ideally use Vector3.FromArrayToRef on the existing `position` and `scaling` vectors instead of allocating new ones. Vector3.Zero, Quaternion.Identity, Vector3.One also allocate which are also unnecessary here.

Reply:
Ok, I needed to implement `Quaternion.FromArrayToRef` to support this behavior. Previously we did not have a way to set a quaternion's values via indexing into an array.